tests/integration/test_config.py

```diff
@@ -339,3 +378,381 @@ async def test_host_group_config(cluster: Cluster) -> None:
+
+    remote_cfg = await get_object_config(obj1, httpx_client)
+    assert remote_cfg[field1] == value3 == cfg2[field1, Parameter].value
+    assert remote_cfg[field2] == initial2 == cfg2[field2, Parameter].value
+    assert remote_cfg[field3] == initial3 == cfg2[field3, Parameter].value
+
+
+async def two_sessions_case_5(obj1: Service, obj2: Service, httpx_client: AsyncClient) -> None:
+    """
+    user 1: set field1 - value1, field2 - value2, field3 - value3;
+    user 2: set field1 - value3, field2 - value2, field3 - value4, save config
+    user 1: refresh (apply remote)
+    """
+    field1, field2, field3 = "complexity_level", "int_field", "int_field2"
+    initial1, initial2, initial3 = 5, None, None
+    value1, value2, value3, value4 = 20, 30, 40, 50
+
+    remote_cfg = await get_object_config(obj1, httpx_client)
+    assert remote_cfg[field1] == initial1
+    assert remote_cfg[field2] == initial2
+    assert remote_cfg[field3] == initial3
+
+    cfg1, cfg2 = await refresh_and_get_two_configs(obj1, obj2)
+    assert isinstance(cfg1, ObjectConfig) and isinstance(cfg2, ObjectConfig)
+
+    # user 1
+    cfg1[field1, Parameter].set(value1)
+    cfg1[field2, Parameter].set(value2)
+    cfg1[field3, Parameter].set(value3)
+
+    # user 2
+    cfg2[field1, Parameter].set(value3)
+    cfg2[field2, Parameter].set(value2)
+    cfg2[field3, Parameter].set(value4)
+    await cfg2.save()
+
+    # user 1
+    await cfg1.refresh(strategy=apply_remote_changes)
+    remote_cfg = await get_object_config(obj1, httpx_client)
+
+    assert cfg1[field1, Parameter].value == value3 == remote_cfg[field1]
+    assert cfg1[field2, Parameter].value == value2 == remote_cfg[field2]
+    assert cfg1[field3, Parameter].value == value4 == remote_cfg[field3]
+
+
+async def two_sessions_case_6(obj1: Service, obj2: Service, host: Host) -> None:
+    """
+    user 1: map host to chg1;
+    user 2: map host to chg2, get error
+    """
+
+    chg1 = await obj1.config_host_groups.create(name="Service CHG 1")
+    chg2 = await obj2.config_host_groups.create(name="Service CHG 2")
+    assert id(chg1._requester) != id(chg2._requester)
+
+    # user 1
+    await chg1.hosts.add(host=host)
+
+    # user 2
+    with pytest.raises(ExceptionGroup) as exc:
+        await chg2.hosts.add(host=host)
+    assert exc.group_contains(ConflictError, match="GROUP_CONFIG_HOST_ERROR")
+
+
+async def two_sessions_case_7(obj1: Service, obj2: Service, host: Host) -> None:
+    """
+    cluster with service, component, host mapped to component, service AHG with the host
+    user 1: remove host from component, save mapping, check service AHG has no hosts;
+    user 2: create new service AHG, map host to this AHG, get error
+    """
+
+    # prepare
+    component = await obj1.components.get(name__eq="component1")
+    mapping = await obj1.cluster.mapping
+    mapping.empty()
+    await mapping.save()
+    await mapping.add(component=component, host=host)
```

Comment:
it's about mapping, not config

imo it should be in a different module

Reply:
moved to test_mapping.py